packages/block-library/src/button/edit.native.js

```diff
@@ -0,0 +1,407 @@
+		} );
+
+		this.setState( { isLinkSheetVisible: false } );
+	}
+
+	render() {
+		const { attributes, textColor, isSelected, clientId } = this.props;
+		const {
+			placeholder,
+			text,
+			borderRadius,
+			url,
+			linkTarget,
+			rel,
+		} = attributes;
+		const { maxWidth, isLinkSheetVisible, isButtonFocused } = this.state;
+
+		const borderRadiusValue = borderRadius !== undefined ? borderRadius : styles.button.borderRadius;
+		const outlineBorderRadius = borderRadiusValue > 0 ? borderRadiusValue + styles.button.paddingTop + styles.button.borderWidth : 0;
+
+		// To achieve proper expanding and shrinking `RichText` on iOS, there is a need to set a `minWidth`
+		// value at least on 1 when `RichText` is focused or when is not focused, but `RichText` value is
+		// different than empty string.
+		const minWidth = isButtonFocused || ( ! isButtonFocused && text && text !== '' ) ? 1 : styles.button.minWidth;
+		// To achieve proper expanding and shrinking `RichText` on Android, there is a need to set
+		// a `placeholder` as an empty string when `RichText` is focused,
+		// because `AztecView` is calculating a `minWidth` based on placeholder text.
+		const placeholderText = isButtonFocused || ( ! isButtonFocused && text && text !== '' ) ? '' : ( placeholder || __( 'Add text…' ) );
+
+		return (
+			<View
+				style={ { flex: 1 } }
+				onLayout={ this.onLayout }
+			>
+				<View
+					style={ [
+						styles.container,
+						isSelected && {
+							borderColor: this.getBackgroundColor(),
+							borderRadius: outlineBorderRadius,
+							borderWidth: styles.button.borderWidth,
+						},
+					] }
+				>
+					<ColorBackground
+						borderRadiusValue={ borderRadiusValue }
+						backgroundColor={ this.getBackgroundColor() }
+					>
+						<RichText
+							setRef={ ( richText ) => {
+								this.richTextRef = richText;
+							} }
+							placeholder={ placeholderText }
+							value={ text }
+							onChange={ this.onChangeText }
+							style={ {
+								...richTextStyle.richText,
+								color: textColor.color || '#fff',
+							} }
+							textAlign="center"
+							placeholderTextColor={ 'lightgray' }
+							identifier="content"
+							tagName="p"
+							minWidth={ minWidth }
+							maxWidth={ maxWidth }
+							id={ clientId }
+							isSelected={ isButtonFocused }
+							withoutInteractiveFormatting
+							unstableOnFocus={ () => this.onToggleButtonFocus( true ) }
+							__unstableMobileNoFocusOnMount={ ! isSelected }
+						/>
+					</ColorBackground>
+
+					{ isButtonFocused && <BlockControls>
+						<ToolbarGroup>
+							<ToolbarButton
+								title={ __( 'Edit image' ) }
+								icon={ 'admin-links' }
+								onClick={ this.onToggleLinkSettings }
+							/>
+						</ToolbarGroup>
+					</BlockControls> }
+
+					<BottomSheet
+						isVisible={ isLinkSheetVisible }
+						onClose={ this.onToggleLinkSettings }
+						hideHeader
+					>
+						{ /* eslint-disable jsx-a11y/no-autofocus */
+							<BottomSheet.Cell
```

Comment:
This looks duplicated. Can't we reuse the same TextControl instance here? or at least the same props.

```
<TextControl
								label={ __( 'Button URL' ) }
								value={ url || '' }
								valuePlaceholder={ __( 'Add URL' ) }
								onChange={ this.onChangeURL }
								autoCapitalize="none"
								autoCorrect={ false }
								separatorType="fullWidth"
								keyboardType="url"
							/>
```

Reply:
As far as I understand we can reuse the `Link Settings` section to have the same code in both places. Generally, I'm into it, however, there are some differences:

* The main difference here is a naming convention: within a `InspectorControls` we have `(control name)Control` and inside `BottomSheet` there are `BottomSheet.(cell name)`, so reusing them means that we'll have to mix it.

* Different label for link relation: `Add rel` vs `Link rel`. I think it should be unified if we want to reuse the code.

* A settings order are different:

![image](https://user-images.githubusercontent.com/22746080/73671932-4b4edc00-46ac-11ea-853b-25d7f6493da7.png)

* Link settings have icons whereby Button options don't have

Reply:
I was imagining something like returning each `<XControl>` instance from a function and pass parameters to make differences possible. It'd be enough to only do this for common ones. 

-------------

>Different label for link relation: Add rel vs Link rel. I think it should be unified if we want to reuse the code.

It is technically possible to keep this difference, however, I doubt this is an intended thing? @iamthomasbishop

Reply:
Yup, I would like to have the same label in both places (`Add rel` or `Link rel` and the same order in settings:
1. Button URL
2. Add rel / Link rel
3. Open in new tab

Reply:
I would use `Link rel` and the order as:

1. Button URL
2. Open in new tab
3. Link rel